var/spack/repos/builtin/packages/fortrilinos/package.py

```diff
@@ -72,3 +74,27 @@ def cmake_args(self):
             self.define('ForTrilinos_EXAMPLES', self.run_tests),
             self.define('ForTrilinos_TESTING', self.run_tests),
         ]
+
+    examples_src_dir = 'example/test-installation'
+
+    @run_after('install')
+    def setup_smoke_tests(self):
+        """Copy the example source files after the package is installed to an
+        install test subdirectory for use during `spack test run`."""
+        self.cache_extra_test_sources([self.examples_src_dir])
+
+    def test(self):
+        example_src_dir = join_path(self.install_test_root,
+                                    self.examples_src_dir)
+        test_build_dir = join_path(self.test_suite.current_test_data_dir,
```

Comment:
The `self.test_suite.current_test_data_dir` only applies when the package has custom files present in the package's directory within the Spack repo so using it in this package could cause confusion if anyone uses it as an example for their package(s).

Since #23713 isn't merged yet, perhaps use `self.test_suite.stage` in the interim?

Reply:
OK, `self.test_suite.stage` seems like a more appropriate choice anyway. Thanks!